src/api/app/controllers/request_controller.rb

```diff
@@ -109,12 +109,16 @@ def update
 
     Suse::Validator.validate(:request, body)
 
+    parsed_xml = Xmlhash.parse(body)
+
+    raise SaveError, 'Failed parsing the request xml' unless parsed_xml
```

Comment:
Did you try if it is not already parsed?
https://github.com/openSUSE/open-build-service/blob/master/src/api/config/initializers/wrap_parameters.rb#L61

Reply:
And I really don't like that you always need to add the SaveError now because next person who will use this function will forget. Why not just leave it in the method?

Reply:
@ChrisBr:

> Did you try if it is not already parsed?
> https://github.com/openSUSE/open-build-service/blob/master/src/api/config/initializers/wrap_parameters.rb#L61

Not yet. I will try.

> And I really don't like that you always need to add the SaveError now because next person who will use this function will forget. Why not just leave it in the method?

When you say 'the method' I'm fine with adding the `raise SaveError` guard clause to the `Xmlhash.parse` method.

If you refer to add the `raise SaveError` guard clause to the `new_from_hash` method, how would you implement the check of the `id` element in the `request_create` method in `src/api/app/controllers/request_controller.rb`?